src/app/profile/yourGiving/yourGiving.component.js

```diff
@@ -0,0 +1,121 @@
+import angular from 'angular';
+import 'angular-ui-bootstrap';
+import range from 'lodash/range';
+import map from 'lodash/map';
+import displayAddress from 'common/components/display-address/display-address.component';
+import givingRecipientView from './givingRecipientView/givingRecipientView.component';
+import givingMonthlyView from './givingMonthlyView/givingMonthlyView.component';
+import includes from 'lodash/includes';
+import loadingComponent from 'common/components/loading/loading.component';
+import loadingOverlay from 'common/components/loadingOverlay/loadingOverlay.component';
+import profileService from 'common/services/api/profile.service';
+import sessionEnforcerService from 'common/services/session/sessionEnforcer.service';
+import sessionService from 'common/services/session/session.service';
+import template from './yourGiving.tpl';
+
+import {Roles} from 'common/services/session/session.service';
+
+let componentName = 'yourGiving';
+
+export const queryParams = {
+  view: 'view'
+};
+
+export const givingViews = ['recipient', 'monthly'];
+
+class YourGivingController {
+
+  /* @ngInject */
+  constructor( $window, $location, $filter, sessionEnforcerService, profileService, sessionService ) {
+    this.$window = $window;
+    this.$location = $location;
+    this.sessionEnforcerService = sessionEnforcerService;
+    this.profileService = profileService;
+    this.sessionService = sessionService;
+    this.dateFilter = $filter( 'date' );
+  }
+
+  $onInit() {
```

Comment:
This seems like a bit of code for `$onInit`... Do you think it would make sense to move it to its own method so it can be tested separately? I guess it is mostly initialization. Maybe you could make `initSessionEnforcer` and `initFilters` methods. Idk. Whatever you think.